k2/python/tests/rmepsilon_test.py

```diff
@@ -0,0 +1,106 @@
+        self.weights = k2.FloatArray1(weights)
+
+    def test_max_weight(self):
+        forward_max_weights = k2.DoubleArray1.create_array_with_size(
+            self.num_states)
+        backward_max_weights = k2.DoubleArray1.create_array_with_size(
+            self.num_states)
+        wfsa = k2.WfsaWithFbWeights(self.fsa, self.weights,
+                                    k2.FbWeightType.kMaxWeight,
+                                    forward_max_weights, backward_max_weights)
+        beam = 8.0
+        remover = k2.EpsilonsRemoverMax(wfsa, beam)
+        fsa_size = k2.IntArray2Size()
+        arc_derivs_size = k2.IntArray2Size()
+        remover.get_sizes(fsa_size, arc_derivs_size)
+        fsa_out = k2.Fsa.create_fsa_with_size(fsa_size)
+        arc_derivs = k2.IntArray2.create_array_with_size(arc_derivs_size)
+        arc_weights_out = k2.FloatArray1.create_array_with_size(fsa_size.size2)
+        remover.get_output(fsa_out, arc_weights_out, arc_derivs)
+        self.assertTrue(k2.is_epsilon_free(fsa_out))
+        self.assertEqual(fsa_out.size1, 6)
+        self.assertEqual(fsa_out.size2, 11)
+        self.assertEqual(arc_derivs.size1, 11)
+        self.assertEqual(arc_derivs.size2, 18)
+        self.assertTrue(
+            k2.is_rand_equivalent_max_weight(self.fsa, self.weights, fsa_out,
+                                             arc_weights_out, beam))
+
+    def test_logsum_weight(self):
+        forward_logsum_weights = k2.DoubleArray1.create_array_with_size(
+            self.num_states)
+        backward_logsum_weights = k2.DoubleArray1.create_array_with_size(
+            self.num_states)
+        wfsa = k2.WfsaWithFbWeights(self.fsa, self.weights,
+                                    k2.FbWeightType.kLogSumWeight,
+                                    forward_logsum_weights,
+                                    backward_logsum_weights)
+        beam = 8.0
+        remover = k2.EpsilonsRemoverLogSum(wfsa, beam)
+        fsa_size = k2.IntArray2Size()
+        arc_derivs_size = k2.IntArray2Size()
+        remover.get_sizes(fsa_size, arc_derivs_size)
+        fsa_out = k2.Fsa.create_fsa_with_size(fsa_size)
+        arc_derivs = k2.LogSumArcDerivs.create_arc_derivs_with_size(
+            arc_derivs_size)
+        arc_weights_out = k2.FloatArray1.create_array_with_size(fsa_size.size2)
+        remover.get_output(fsa_out, arc_weights_out, arc_derivs)
+        self.assertTrue(k2.is_epsilon_free(fsa_out))
+        self.assertEqual(fsa_out.size1, 6)
+        self.assertEqual(fsa_out.size2, 11)
+        self.assertEqual(arc_derivs.size1, 11)
+        self.assertEqual(arc_derivs.size2, 20)
+        self.assertTrue(
+            k2.is_rand_equivalent_after_rmeps_pruned_logsum(
+                self.fsa, self.weights, fsa_out, arc_weights_out, beam))
+        # cast float to int
+        arc_ids = k2.StridedIntArray1.from_float_tensor(arc_derivs.data[:, 0])
+        self.assertEqual(arc_ids.get_data(1), 1)
```

Comment:
@danpovey here I reinterpret `tensor.FloatArray` (i.e. `arc_derivs[:,0]`) as `StridedIntArray1`, so we can pass this `(Strided)IntArray1` object to C++ code that accepts `IntArray1` or `int*`. We can do same thing in the future when create bindings for new struct `Arc` in `cuda_draft`

Reply:
@danpovey  have you reviewed this PR? I updated it last Saturday.